home/views.py

```diff
@@ -92,27 +91,35 @@ class SessionDetailView(DetailView):
     model = Session
     template_name = "home/prerelease/session_detail.html"
 
+    def get_queryset(self):
+        return Session.objects.with_applications(user=self.request.user)
+
 
 class SessionListView(ListView):
     model = Session
     template_name = "home/prerelease/session_list.html"
     context_object_name = "sessions"
 
+    def get_queryset(self):
+        return Session.objects.with_applications(user=self.request.user)
+
 
-@method_decorator(login_required, name="dispatch")
-class CreateUserSurveyResponseFormView(FormMixin, DetailView):
+class CreateUserSurveyResponseFormView(
+    LoginRequiredMixin, UserPassesTestMixin, FormMixin, DetailView
+):
     model = Survey
     object = None
     form_class = CreateUserSurveyResponseForm
     success_url = reverse_lazy("session_list")
     template_name = "home/surveys/form.html"
 
-    def dispatch(self, request, *args, **kwargs):
+    def test_func(self):
         survey = self.get_object()
-        if UserSurveyResponse.objects.filter(survey=survey, user=request.user).exists():
-            messages.warning(request, gettext("You have already submitted."))
-            return redirect("session_list")
-        return super().dispatch(request, *args, **kwargs)
+        user = self.request.user
+        return (
+            user.profile.email_confirmed
+            and not UserSurveyResponse.objects.filter(survey=survey, user=user).exists()
```

Comment:
I understand why we need this, but I think it belongs in `CreateUserSurveyResposneForm.clean()` so it could be rendered in  [`non_field_errors`](https://docs.djangoproject.com/en/5.0/ref/forms/api/#django.forms.Form.non_field_errors). I think a 403  error (I'm assuing that's what UserPassesTestMixin returns) is less obvious to the user why their submission was rejected.

Reply:
Ah that would mean they fill out the form again and submit and learn they're not allowed?
I think eventually I'll have a way for them to see their submissions and it would be available to them

Reply:
Fair enough!